### PR TITLE
Adding 4 new loss functions

### PR DESCRIPTION
**Pull request by hamidriasat**
Following changes are included in current PR

- Four new loss functions added
          1.   Jaccard/IoU loss
          2.   SSIM loss
          3.   Unet3+ loss
          4.   BASNet loss
- Hybrid losses implemented in a separate file
- `summary.png` removed instead, table is added
- README updated

### CODE REVIEW

**Review by shruti-jadon — COMMENTED** [review]
Add hybrid losses functions to loss_functions.py to maintain same loss functions file.

**Review by shruti-jadon — COMMENTED** [review]
Don't delete summary.png

File: example_usage/run.py [hunk 1]
```diff
@@ -102,18 +103,21 @@ def my_generator(x_train, y_train, batch_size):
...
+semantic_loss = Semantic_loss_functions()
+hybrid_loss = HybridLossFunctions()
 
-s = Semantic_loss_functions()
 model = get_small_unet_no_pool()
-model.compile(optimizer=Adam(lr=1e-3), loss=s.focal_tversky,
-              metrics=[s.dice_coef, s.sensitivity, s.specificity])
+model.compile(optimizer=Adam(lr=1e-3),
+              # loss=semantic_loss.basnet_hybrid_loss,
+              loss=hybrid_loss.basnet_hybrid_loss,
```
Comment (shruti-jadon) [inline example_usage/run.py]: move all new functions to loss_functions.py, call the basnet_hybrid_loss from semantic_loss/s object.

Comment (hamidriasat) [reply example_usage/run.py]: In next commit, I am goanna move all loss functions to `loss_functions.py` file and call them with semantic_loss/s object.

But I cannot add reference of new loss functions to summary.png.  So I guess we have to replace it with table.
If you have any work around in mind, then kindly share it.

Comment (shruti-jadon) [reply example_usage/run.py]: I can see you have updated the README.md, don't delete the summar.png, it is anyways not being used in README.md after your changes.

**Comment by hamidriasat** [comment]
As per new requirements, following changes have been made in current commit.

- Hybrid loss functions moved from `hybrid_losses.py` to `loss_functions.py`
- hybrid losses tested using `example_usage/run.py`
- `README.md` updated
- `summary.png` file added again

**Review by shruti-jadon — APPROVED** [review]
